util/src/main/java/com/google/cloud/hadoop/util/HttpTransportFactory.java

```diff
@@ -165,4 +171,56 @@ static URI parseProxyAddress(@Nullable String proxyAddress) {
           String.format("Invalid proxy address '%s'.", proxyAddress), e);
     }
   }
+
+  static class KeepAliveSslSocketFactory extends SSLSocketFactory {
+
+    private final SSLSocketFactory wrappedSockedFactory;
+
+    public KeepAliveSslSocketFactory() {
+      wrappedSockedFactory = HttpsURLConnection.getDefaultSSLSocketFactory();
+    }
+
+    @Override
+    public String[] getDefaultCipherSuites() {
+      return wrappedSockedFactory.getDefaultCipherSuites();
+    }
+
+    @Override
+    public String[] getSupportedCipherSuites() {
+      return wrappedSockedFactory.getSupportedCipherSuites();
+    }
+
+    @Override
+    public Socket createSocket(Socket s, String host, int port, boolean autoClose)
+        throws IOException {
+      return addExtraParams(wrappedSockedFactory.createSocket(s, host, port, autoClose));
+    }
+
+    public Socket createSocket(String host, int port) throws IOException {
+      return addExtraParams(wrappedSockedFactory.createSocket(host, port));
+    }
+
+    public Socket createSocket(InetAddress address, int port) throws IOException {
+      return addExtraParams(wrappedSockedFactory.createSocket(address, port));
+    }
+
+    public Socket createSocket(String host, int port, InetAddress clientAddress, int clientPort)
+        throws IOException {
+      return addExtraParams(
+          wrappedSockedFactory.createSocket(host, port, clientAddress, clientPort));
+    }
+
+    public Socket createSocket(
+        InetAddress address, int port, InetAddress clientAddress, int clientPort)
+        throws IOException {
+      return addExtraParams(
+          wrappedSockedFactory.createSocket(address, port, clientAddress, clientPort));
+    }
+
+    private static Socket addExtraParams(Socket socket) throws SocketException {
+      socket.setKeepAlive(true);
+      socket.setSoTimeout(200_000);
```

Comment:
We need to make this configurable. How is this timeout different/interact with other timeouts that we already set?
https://github.com/GoogleCloudDataproc/hadoop-connectors/blob/9f9d5f821b534689ddb1f1c3754c0414fa2483a4/util/src/main/java/com/google/cloud/hadoop/util/RetryHttpInitializer.java#L245-L246

Reply:
Timeout doesn't need to be configured, it will take the default or the one configured at OS level. Removed.